dora/core/common/src/main/java/alluxio/underfs/BaseUfsStatusIterator.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.underfs;
+
+import alluxio.collections.Pair;
+import alluxio.underfs.options.ListOptions;
+import alluxio.util.io.PathUtils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.LinkedList;
+
+/**
+ * This base under file system status iterator is for listing files iteratively.
+ */
+public class BaseUfsStatusIterator implements Iterator<UfsStatus> {
```

Comment:
is this used for listing the statuses of **files only**? If so, can rename it to `UfsFileStatusIterator` for clarity.

Reply:
Also the iterator can be made of type `Iterator<UfsFileStatus>`

Reply:
I renamed the class to `UfsFileStatusIterator`. But changing the type to `Iterator<UfsFileStatus>` may break too many codes. Let's still use `Iterator<UfsStatus>` in this PR.